packages/datepickers/src/styled/StyledDay.ts

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import styled, { DefaultTheme, ThemeProps } from 'styled-components';
+import { retrieveComponentStyles, getColor, DEFAULT_THEME } from '@zendeskgarden/react-theming';
+
+interface IStyledDayProps extends ThemeProps<DefaultTheme> {
+  isPreviousMonth?: boolean;
+  isToday?: boolean;
+  isDisabled?: boolean;
+  isSelected?: boolean;
+  isSmall: boolean;
+}
+
+const retrieveStyledDayColor = ({
+  isDisabled,
+  isSelected,
+  isToday,
+  isPreviousMonth,
+  theme
+}: IStyledDayProps) => {
+  if (isDisabled) {
+    return getColor('neutralHue', 400, theme);
+  }
+
+  if (isSelected && !isDisabled) {
+    return theme.colors.background;
+  }
+
+  if (isToday) {
+    return 'inherit';
+  }
+
+  if (isPreviousMonth) {
+    return getColor('neutralHue', 600, theme);
+  }
+
+  return getColor('primaryHue', 600, theme);
+};
+
+const retrieveBackgroundColor = ({ isSelected, isDisabled, theme }: IStyledDayProps) => {
+  if (isSelected && !isDisabled) {
+    return getColor('primaryHue', 600, theme);
+  }
+
+  return 'inherit';
+};
+
+const COMPONENT_ID = 'datepickers.day';
+
+const StyledDay = styled.div.attrs({
+  'data-garden-id': COMPONENT_ID
+})<IStyledDayProps>`
+  display: flex;
+  position: absolute;
+  align-items: center;
+  justify-content: center;
+  border-radius: 50%;
+  background-color: ${retrieveBackgroundColor};
+  cursor: ${props => (props.isDisabled ? 'inherit' : 'pointer')};
```

Comment:
There shouldn't be an `isDisabled` prop here (or throughout `datepickers`). Use `[aria-disabled]` if applying `:disabled` styling to html element that does not accept the `disabled` attribute.

Reply:
From our offline conversation, the combination of the states and `isDisabled` logic make this difficult to read with an attribute.

I have refactored this area to a helper function to help with readability.